frodo.js

```diff
@@ -2,68 +2,65 @@
  * Extras Plugin
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.